src/Common/filesystemHelpers.cpp

```diff
@@ -122,6 +123,17 @@ bool pathStartsWith(const std::filesystem::path & path, const std::filesystem::p
     return path_starts_with_prefix_path;
 }
 
+bool symlinkStartsWith(const std::filesystem::path & path, const std::filesystem::path & prefix_path)
+{
+    auto absolute_path = std::filesystem::absolute(path);
```

Comment:
The point is not to use fs::canonical or fs::weakly_canonical

Reply:
I feel like I need a comment to:
* function `symlinkStartsWith` itself - about how it works with symlinks
* probably, every call for `std::filesystem`, cause I need to go to documentation to understand what exactly it does; and also why it is needed here

Reply:
BTW, can't we just canonise both path and then compare?

Reply:
> can't we just canonise both path and then compare?

This is exactly what was done and why it did not work. Applying fs::canonical or fs::weakly_canonical to a symlink results in a canonical path -- where it links to, but what we need is a path of the symlink itself.